pkg/executor/internal/mpp/recovery_handler.go

```diff
@@ -165,35 +158,32 @@ func (*memLimitHandlerImpl) doRecovery(info *RecoveryInfo) error {
 }
 
 type mppResultHolder struct {
-	capacity uint64
-	// True when holder is full or begin to return result.
-	cannotHold bool
-	curRows    uint64
-	chks       []*chunk.Chunk
 	memTracker *memory.Tracker
+	resps      []*mppResponse
+	capacity   uint64
+	cannotHold bool
 }
 
 func newMPPResultHolder(holderCap uint64, parent *memory.Tracker) *mppResultHolder {
 	return &mppResultHolder{
 		capacity:   holderCap,
-		chks:       []*chunk.Chunk{},
+		resps:      []*mppResponse{},
```

Comment:
init `resps` using `holderCap` as the reserved cap?

Reply:
done